src/mscorlib/shared/System/Globalization/GregorianCalendar.cs

```diff
@@ -520,8 +520,13 @@ public override DateTime ToDateTime(int year, int month, int day, int hour, int
             throw new ArgumentOutOfRangeException(nameof(era), SR.ArgumentOutOfRange_InvalidEraValue);
         }
 
-        internal override Boolean TryToDateTime(int year, int month, int day, int hour, int minute, int second, int millisecond, int era, out DateTime result)
+        internal override bool TryToDateTime(int year, int month, int day, int hour, int minute, int second, int millisecond, int era, out DateTime result)
         {
+            if (!IsValidDay(year, month, day, era))
+            {
+                result = DateTime.Now;
```

Comment:
Previously, if era != CurrentEra && era != ADEra it would set result = DateTime.MinValue, wheras this will potentially return DateTime.Now.

DateTime.Now is much more expensive than DateTime.MinValue so ideally we could return the latter. It's true the code currently will return DateTime.Now in some failure cases. But this seems contrary to the [docs ](https://msdn.microsoft.com/en-us/library/ch92fbc1(v=vs.110).aspx) which say "returns MinValue if the conversion failed."

I suggest this always set DateTime.MinValue in failure cases, @tarekgh?